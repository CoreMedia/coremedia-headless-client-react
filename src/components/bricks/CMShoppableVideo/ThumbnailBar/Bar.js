// @flow
import styled from 'styled-components';
import theme from 'styled-theming';

const color = theme('name', {
  default: 'rgba(0, 108, 174, 1)',
  hkm: '#fabfc0',
});

const Bar = styled.hr.attrs({
  'aria-hidden': true,
})`
  position: relative;
  left: ${props => `${props.posX}px`};
  width: ${props => `${props.width}px`};
  margin: 0;
  border-width: 0 0 3px;
  border-style: none none solid;
  border-color: ${props => (props.width ? color : 'transparent')};
  transition: ${props => (props.posX && props.width ? 'all' : 'border-color')} 0.6s ease-out 0s;
`;
Bar.displayName = 'Bar';

/*Bar.propTypes = {
  posX: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  theme: PropTypes.shape({
    name: PropTypes.string.isRequired,
  }),
};*/

Bar.defaultProps = {
  theme: {
    name: 'default',
  },
};

export default Bar;
