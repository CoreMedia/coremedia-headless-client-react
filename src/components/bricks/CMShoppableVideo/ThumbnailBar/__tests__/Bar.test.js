// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Bar from '../Bar';

describe('Bar Component', () => {
  describe('with default theme', () => {
    it('should render correctly with zero width', () => {
      const tree = renderer.create(<Bar posX={0} width={0} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render correctly with 10px width', () => {
      const tree = renderer.create(<Bar posX={0} width={10} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render correctly with 10px width and 5px posX', () => {
      const tree = renderer.create(<Bar posX={5} width={10} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('with hkm theme', () => {
    it('should render correctly with zero width', () => {
      const tree = renderer.create(<Bar theme={{ name: 'hkm' }} posX={0} width={0} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render correctly with 10px width', () => {
      const tree = renderer.create(<Bar theme={{ name: 'hkm' }} posX={0} width={10} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render correctly with 10px width and 5px posX', () => {
      const tree = renderer.create(<Bar theme={{ name: 'hkm' }} posX={5} width={10} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
