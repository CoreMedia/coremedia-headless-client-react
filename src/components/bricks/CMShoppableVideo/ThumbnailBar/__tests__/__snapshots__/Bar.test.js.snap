// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Component with default theme should render correctly with 10px width 1`] = `
.c0 {
  position: relative;
  left: 0px;
  width: 10px;
  margin: 0;
  border-width: 0 0 3px;
  border-style: none none solid;
  border-color: rgba(0,108,174,1);
  -webkit-transition: border-color 0.6s ease-out 0s;
  transition: border-color 0.6s ease-out 0s;
}

<hr
  aria-hidden={true}
  className="c0"
  width={10}
/>
`;

exports[`Bar Component with default theme should render correctly with 10px width and 5px posX 1`] = `
.c0 {
  position: relative;
  left: 5px;
  width: 10px;
  margin: 0;
  border-width: 0 0 3px;
  border-style: none none solid;
  border-color: rgba(0,108,174,1);
  -webkit-transition: all 0.6s ease-out 0s;
  transition: all 0.6s ease-out 0s;
}

<hr
  aria-hidden={true}
  className="c0"
  width={10}
/>
`;

exports[`Bar Component with default theme should render correctly with zero width 1`] = `
.c0 {
  position: relative;
  left: 0px;
  width: 0px;
  margin: 0;
  border-width: 0 0 3px;
  border-style: none none solid;
  border-color: transparent;
  -webkit-transition: border-color 0.6s ease-out 0s;
  transition: border-color 0.6s ease-out 0s;
}

<hr
  aria-hidden={true}
  className="c0"
  width={0}
/>
`;

exports[`Bar Component with hkm theme should render correctly with 10px width 1`] = `
.c0 {
  position: relative;
  left: 0px;
  width: 10px;
  margin: 0;
  border-width: 0 0 3px;
  border-style: none none solid;
  border-color: #fabfc0;
  -webkit-transition: border-color 0.6s ease-out 0s;
  transition: border-color 0.6s ease-out 0s;
}

<hr
  aria-hidden={true}
  className="c0"
  width={10}
/>
`;

exports[`Bar Component with hkm theme should render correctly with 10px width and 5px posX 1`] = `
.c0 {
  position: relative;
  left: 5px;
  width: 10px;
  margin: 0;
  border-width: 0 0 3px;
  border-style: none none solid;
  border-color: #fabfc0;
  -webkit-transition: all 0.6s ease-out 0s;
  transition: all 0.6s ease-out 0s;
}

<hr
  aria-hidden={true}
  className="c0"
  width={10}
/>
`;

exports[`Bar Component with hkm theme should render correctly with zero width 1`] = `
.c0 {
  position: relative;
  left: 0px;
  width: 0px;
  margin: 0;
  border-width: 0 0 3px;
  border-style: none none solid;
  border-color: transparent;
  -webkit-transition: border-color 0.6s ease-out 0s;
  transition: border-color 0.6s ease-out 0s;
}

<hr
  aria-hidden={true}
  className="c0"
  width={0}
/>
`;
